### PR TITLE
feat!: add transport api [WPB-12119] remove validation callbacks [WPB-14463]

### PR DESCRIPTION
**Pull request by SimonThormeyer**
# What's new in this PR

----
##### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: crypto-ffi/src/wasm/mod.rs [hunk 1]
```diff
@@ -1327,133 +1334,137 @@ impl log::Log for CoreCryptoWasmLogger {
...
+    fn from(response: WasmMlsTransportResponse) -> Self {
+        match response.variant {
+            MlsTransportResponseVariant::Success => MlsTransportResponse::Success,
+            MlsTransportResponseVariant::Retry => MlsTransportResponse::Retry,
+            MlsTransportResponseVariant::Abort => {
+                if let Some(data) = response.data {
+                    MlsTransportResponse::Abort { reason: data }
+                } else {
+                    MlsTransportResponse::Abort { reason: String::new() }
+                }
```
Comment (coriolinus) [inline crypto-ffi/src/wasm/mod.rs:1362]:
```suggestion
                MlsTransportResponse::Abort { reason: response.data.unwrap_or_default() }
```

File: crypto-ffi/src/wasm/mod.rs [hunk 2]
```diff
@@ -1327,133 +1334,137 @@ impl log::Log for CoreCryptoWasmLogger {
...
+                variant: MlsTransportResponseVariant::Success,
+                data: None,
+            },
+            MlsTransportResponse::Retry => WasmMlsTransportResponse {
+                variant: MlsTransportResponseVariant::Retry,
+                data: None,
+            },
+            MlsTransportResponse::Abort { reason } => WasmMlsTransportResponse {
+                variant: MlsTransportResponseVariant::Abort,
+                data: Some(reason),
```
Comment (coriolinus) [inline crypto-ffi/src/wasm/mod.rs:1381]:
```suggestion
                data: (!reason.is_empty()).then_some(reason),
```

File: crypto/src/context.rs [hunk 3]
```diff
@@ -88,22 +88,26 @@ impl CentralContext {
         }
     }
 
-    pub(crate) async fn callbacks(
-        &self,
-    ) -> CryptoResult<RwLockReadGuardArc<Option<Arc<dyn CoreCryptoCallbacks + 'static>>>> {
+    // This is going to be needed soon.
+    #[expect(dead_code)]
```
Comment (coriolinus) [inline crypto/src/context.rs:92]: Is this still dead code?

File: crypto/src/context.rs [hunk 4]
```diff
@@ -88,22 +88,26 @@ impl CentralContext {
...
-            ContextState::Valid { callbacks, .. } => Ok(callbacks.read_arc().await),
+            ContextState::Valid {
+                transport: callbacks, ..
+            } => Ok(callbacks.read_arc().await),
             ContextState::Invalid => Err(CryptoError::InvalidContext),
         }
     }
 
+    // This is going to be needed soon.
+    #[expect(dead_code)]
```
Comment (coriolinus) [inline crypto/src/context.rs:103]: This too

**Review by coriolinus — APPROVED** [review]

File: crypto/src/context.rs (hunk 3 above)
Comment (SimonThormeyer) [reply crypto/src/context.rs:92]: Yep, otherwise this would throw an error :) (that's the difference to `#[allow()]`)

**Pull request merged**